### DISCUSSION TITLE
Encode time from UNIX to CUC

### DISCUSSION THREAD

**Discussion by Stavrenas** [yamcs/yamcs#710] [Q&A] [answered]
Hello everyone!
I could not find anything in documentation regarding the encoding`AbsoluteTimeArgumentType`. We need to be able to encode the time from UNIX to a CUC format but the calibrator does not seem to do anything.  Currently , the argument is defined as: 

```
        <xtce:AbsoluteTimeArgumentType name="release_time_calendar">
            <xtce:Encoding>
                <xtce:IntegerDataEncoding sizeInBits="32">
                <xtce:DefaultCalibrator>
                    <xtce:PolynomialCalibrator>
                        <xtce:Term coefficient="-15778296000" exponent="0" />
                        <xtce:Term coefficient="10" exponent="1" />
                    </xtce:PolynomialCalibrator>
                </xtce:DefaultCalibrator>
            </xtce:IntegerDataEncoding>
            </xtce:Encoding>
        <xtce:ReferenceTime>
            <xtce:Epoch>UNIX</xtce:Epoch>
        </xtce:ReferenceTime>
        </xtce:AbsoluteTimeArgumentType> 
```
So , am I missing something? I can't find any examples nor documentation

**Comment by xpromache** [comment]
I didn't know you can have calibrators for absolute time arguments. Is that valid XTCE?

Yamcs does some conversion based on the epoch and offset, see here:
https://github.com/yamcs/yamcs/blob/master/yamcs-core/src/main/java/org/yamcs/mdb/ArgumentTypeProcessor.java#L197

about the lack of doc, true seems to be lacking:
https://docs.yamcs.org/yamcs-server-manual/mdb/data-types/#absolute-time-data-type

help is most welcome...

**Reply by Stavrenas** [reply]
Yes, I also found this file but I'm having trouble integrating it to my yamcs-instance. I do not know how to configure the project files to also inlude ArgumentTypeProcessor. XTCE docs state that AbsoluteTimeParameterType has an encoding field, where we can insert an integer encoding element and thus a calibrator. But it does not work.

**Reply by Stavrenas** [reply]
So, from what I understand, ArgymentTypeProcessor is a source file. How can I edit this and compile Yamcs with it?

**Comment by xpromache** [comment, marked as answer]
That is part of Yamcs. It uses it when converting a time argument to raw value. I showed it so you can see what is it doing. Try defining a argument with an epoch and offset instead of a calibrator.

**Reply by Stavrenas** [reply]
You are right. I figured it out. Thank you very much !
